### PR TITLE
Site Editor: update index view for pages

### PR DESCRIPTION
**Pull request by oandregal**
Part of https://github.com/WordPress/gutenberg/issues/59659
Related https://github.com/WordPress/gutenberg/pull/59792 and https://github.com/WordPress/gutenberg/issues/55083

## What?

Makes the DataViews-powered Page the index view for the pages section.

https://github.com/WordPress/gutenberg/assets/583546/3c0a4e9e-e0da-4589-804e-f948a9b59de4

## Why?

It's the direction we want to go.

## How?

- Update sidebar area cc993b5dbfaa782109e8b41696878194f2c16ecf
- Delete old sidebar 3ac9349f7c420844c3278aba7104ff195bb0efc6
- Update content area 05af5c988a934a24903270261460f79f9a483e6a
- Update default layout to list a2a373a10be65add43b8d743524976f98e023866
- Update preview area bf846c460f8caa6479786e9230a28224c6d8d351
- Clear params for page path e5f7e7dbaf9798ddcea12ca7cb692504b0353f16

## Testing Instructions

Go to the Site Editor and visit the Pages section. It should display the list layout for pages.

## Follow-ups

I've ran into the following issues while testing. This also happens in `trunk` so I plan to investigate separately:

- The list layout (content&preview) don't use the URL params to update selection on first load.
- For custom views, the list layout doesn't show the preview.
- When it goes to the editor, it uses `path=page` instead of `postType=page`. The side-effect is that when coming back, it has the `path` and goes directly to list layout. It should go to the details view for the time being.

### CODE REVIEW

**Comment by youknowriad** [comment]
I think this highlights a couple things:

 - In "list view" there's no way to go into "details" view. When clicking a preview, you go into the editor and going back brings you back to the list view. I think that's what we want.
 - That said, when switching to either "table" or "grid" views and clicking on pages, going back passes by the details panel first.

Should we try and "link" to the same "url" we use for "list view" in the other layouts too? I'm guessing that means illuminating the "details" panel entirely but to be honest, it's already very hidden with the current state of the PR. Should we wait for the unification of the details and inspector to advance more first?

---

The other thing that this highlights is that our "frame" animation is kind of broken. The "right side" of the frame/preview should stay stable when you navigate from the root of the site editor into "pages".

**Comment by jameskoster** [comment]
I don't think it needs to block this PR but noting that once merged we'll lose the cross-links from the Pages panel to relevant templates like 404 and search results. Related: https://github.com/WordPress/gutenberg/issues/41716

Screenshot 2024-03-18 at 13 39 56

It would be good to have some ideas in mind about how we replicate these in data views. Is it technically feasible to display these templates amongst regular pages? Or would a separate view ('Utility' or similar) within Pages work better?

**Comment by Marc-pi** [comment]
Be carefull, the usage of 14/15/16' laptop screens size is very frequent
i do fear that having 3 panes (2+1) as shown in the screencast will make it difficult to just manage content editing...
not everyone is using a 40' screen like devs or designers !

**Comment by SaxonF** [comment]
How challenging would it be to move the "views" functionality (left sidebar) into its own PR/experiment?

**Comment by SaxonF** [comment]
We also need to add an edit button and the ellipsis menu to rows on hover. Edit button opens editor which is identical to clicking frame.

image

**Comment by oandregal** [comment]
> In "list view" there's no way to go into "details" view. (...) Should we try and "link" to the same "url" we use for "list view" in the other layouts too?

Yes, definitely. I was planning to look into that as a follow-up given that it already works that way in `trunk`, but perhaps it's best to do it in this PR as it changes the main flow — best not to change it too many times.

> I don't think it needs to block this PR but noting that once merged we'll lose the cross-links from the Pages

In the previous iteration we had them in the left sidebar, as they are now in `trunk`. Is that idea still valid? Probably a follow-up if that's ok.

> How challenging would it be to move the "views" functionality (left sidebar) into its own PR/experiment?

Can you expand on this?

In the sidebar area, we have the default views (shipped in 6.5 and stable) plus the custom views (the only piece that remains hidden under the "wp-admin" experiment).

> We also need to add an edit button and the ellipsis menu to rows on hover.

Yes. I'm landing [the whole issue](https://github.com/WordPress/gutenberg/issues/59659) by pieces.

**Comment by jameskoster** [comment]
> Is that idea still valid? Probably a follow-up if that's ok.

It might be okay as a stop-gap, but I hope we can come up with something better. Fine to handle in a follow-up.

**Comment by oandregal** [comment]
@youknowriad I've fixed the back flow and it works like the existing pages in `trunk`:

- if there's a postId parameter: it goes back through the details to the index view
- if there is not: it goes back to the index view directly

Trunk

https://github.com/WordPress/gutenberg/assets/583546/82d99de1-2edb-4506-906d-01c532a1151b

This PR

https://github.com/WordPress/gutenberg/assets/583546/cb9d78d8-da4e-4982-bbac-e082a5da882d

File: packages/edit-site/src/components/layout/router.js
Comment (youknowriad) [inline packages/edit-site/src/components/layout/router.js]: I'm not sure about this change. This has multiple side effects:

 - Adds a wrapper around the editor which probably have CSS impacts
 - A link around a div is probably not valid HTML
 - Most concerning for me: Causes a remount of the "preview/editor" component when you navigate to another route that doesn't have the link wrapper.

We already have an `onClick` handler somewhere within the `Editor` component, so it feels like it should be fixed there instead.

**Comment by youknowriad** [comment]
@oandregal In the last video you just shared, it's really unclear why the behavior is different when you "select item" or not. I feel like they should both do the same and go back to the list view no?

**Comment by oandregal** [comment]
>  I feel like they should both do the same and go back to the list view no?

I actually understood the opposite in your [previous comment](https://github.com/WordPress/gutenberg/pull/59950#issuecomment-2003715731): that we should maintain the details view until the new experience that substitutes it is in place.

This works the same it currently does in `trunk`: when nothing is selected, we don't have information to display the details view (the `postId` param), so it doesn't go through the details view. It's also the same that happens in the site editor root:

https://github.com/WordPress/gutenberg/assets/583546/cb3390b9-0d01-414f-ac50-811e8f295dd1

File: packages/edit-site/src/components/layout/router.js
Comment (oandregal) [reply packages/edit-site/src/components/layout/router.js]: I can look into exposing that `onClick` handler or alternatives. [This approach](https://github.com/WordPress/gutenberg/pull/57938/files#diff-44881c176d8762d8e2b3cb7d8a82d1fe82c69002b50817de1663052ec167a618) was what we had before the router refactoring, when the editor went back to the list view through the details.

Comment (oandregal) [reply packages/edit-site/src/components/layout/router.js]: https://github.com/WordPress/gutenberg/pull/59950/commits/445b150415c777529f4196559fee67c33a0e5f04 exposes the editor's `onClick` handle and uses it to set the URL params.

**Review by youknowriad — APPROVED** [review]
Maybe this is a necessary evil before we complete the merge of "details" and "inspector".

I'll defer to @jameskoster and @SaxonF here.

**Comment by oandregal** [comment]
It was also strange for me to see how the back flow goes through details _only_ if there's a previous selection (`postId` in URL). A potential follow-up is to make the `postId` parameter always part of the URL: set it upon the site editor's first load. We have that info as part of the edited entity, though I understand we are moving towards having a single source of truth for navigation (the URL), so I'd rather do that. Thoughts?

**Comment by youknowriad** [comment]
> A potential follow-up is to make the postId parameter always part of the URL: set it upon the site editor's first load. We have that info as part of the edited entity, though I understand we are moving towards having a single source of truth for navigation (the URL), so I'd rather do that. Thoughts?

I don't like that, I think the issue will fix itself when we remove the details step.

**Comment by jameskoster** [comment]
> In the last video you just shared, it's really unclear why the behavior is different when you "select item" or not. I feel like they should both do the same and go back to the list view no?

I don't fully understand the technical implications but I agree with this. Ideally the W button should take you to list view.

Is the concern that if we take you to list view then the details panel is effectively removed from the experience? For pages that doesn't seem like a huge deal because their detail panels offers very little value. But I suppose it would be a bit problematic for templates like Blog Home, where you can edit settings in the Details panel but not in the Inspector (yet). If we don't want to remove access to those settings then yes this is may be a necessary evil for now.

**Comment by oandregal** [comment]
> I don't fully understand the technical implications but I agree with this. Ideally the W button should take you to list view.

This is how the list view works in `trunk` and how it initially worked in this PR (no details in the back flow).

I changed it to 1) make it similar to how table & grid work, which still use the details and 2) because when I asked around about this flow, people felt it was better to leave the details until the new thing lands (as I understand it, the details has a purpose that hasn't been yet absorbed by any other thing).

If the feeling has changed, I'm happy to get rid of the details entirely, though it's a bit orthogonal to this work.

Going to go ahead with this to unblock further work.

**Pull request merged**
